src/main/scala/mesosphere/marathon/api/akkahttp/SystemController.scala

```diff
@@ -34,29 +33,8 @@ class SystemController(val marathonConfig: MarathonConf, val cfg: Config, val el
   import Directives._
   import EntityMarshallers._
 
-  def acceptsTextPlain(ranges: Seq[MediaRange]): Boolean = {
-    ranges.exists(range => range.matches(MediaTypes.`text/plain`))
-  }
-  def acceptsApplicationJson(ranges: Seq[MediaRange]): Boolean = {
-    ranges.exists(range => range.matches(MediaTypes.`application/json`))
-  }
-  /**
-    * GET /ping
-    * @return a simple pong as text/plain
-    */
-  def ping: Route = extract(_.request.header[Accept]) { maybeAccept: Option[Accept] =>
-    // Match by priority. Always prefer text/plain.
-    maybeAccept match {
-      case None =>
-        complete("pong")
-      case Some(accept) if (acceptsTextPlain(accept.mediaRanges)) =>
-        complete("pong")
-      case Some(accept) if (acceptsApplicationJson(accept.mediaRanges)) =>
-        complete(JsString("ping"))
-      case Some(_) =>
-        complete(HttpResponse(StatusCodes.NoContent))
-    }
-  }
+  def pingPlainText: Route = complete("pong")
+  def pingJson: Route = complete(JsString("ping"))
```

Comment:
What is the default content-type responded here? Do you think we should explicitly stick on some particular type (ex. "text/plain") in case the default changes in the future?

Reply:
the default one is actually specified in route like this. 

(get & acceptsAnything) { pingPlainText } ~

Reply:
Do you think it's possible to do something like this?
```scala
  sealed trait Pong
  case object Pong extends Pong

  implicit val pongMarshaller: ToResponseMarshaller[Pong]  = Marshaller.oneOf(
    Marshaller.combined{case Pong => JsString("pong")},
    Marshaller.StringMarshaller.compose { case Pong => "pong" },
    Marshaller.opaque{case Pong => "pong"},
    Marshaller.withFixedContentType(ContentTypes.NoContentType){ case Pong => HttpResponse(StatusCodes.NoContent)}
  )
```
and then in the route it will be just 
```scala
path("ping") {
  complete(Pong)
}
```

Reply:
it is possible, but I discussed this with @jeschkies before implementing this and he wanted this to be explicitly done via routes...

I don't mind if it is implemented in either way...

I'll merge this one but I am happy to reimplement it using marshallers - or feel free to submit PR and we can discuss it there